core/shared/src/main/scala/fs2/async/async.scala

```diff
@@ -49,20 +49,30 @@ package object async {
     mutable.Queue.circularBuffer[F,A](maxSize)
 
   /**
-   * Converts a discrete stream to a signal. Returns a single-element stream.
+   * Pipe that holds the last seen element of the source stream.
    *
-   * Resulting signal is initially `initial`, and is updated with latest value
-   * produced by `source`. If `source` is empty, the resulting signal will always
-   * be `initial`.
+   * Backed by a discrete or continuous signal, specified via `discrete` parameter.
+   * By default, the returned stream terminates when the source stream terminates, configured
+   * via the `haltWhenSourceHalts` parameter.
    *
-   * @param source   discrete stream publishing values to this signal
+   * @param initial value emitted until first element from source stream is received
+   * @param discrete true if returned stream should be backed by a discrete signal or false if continuous
+   * @param haltWhenSourceHalts true if stream should be terminated when the source stream terminates
    */
```

Comment:
Even though, in light of the discussion on #907, I'm not sure the 0.9 behaviour can be obtained, I feel like this signature is less powerful: you lose the ability the asynchronously "sample" the Stream with `Signal.get`, and also need to know at the start whether you want a continuous or discrete Stream, so you can't share the Stream returned by `hold` with two consumers having different requirements.
Not quite sure how to improve it though.